src/main/java/eu/europeana/metis/sandbox/service/validationworkflow/HarvestValidationStep.java

```diff
@@ -0,0 +1,58 @@
+package eu.europeana.metis.sandbox.service.validationworkflow;
+
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.common.Step;
+import eu.europeana.metis.sandbox.domain.Record;
+import eu.europeana.metis.sandbox.domain.RecordInfo;
+import eu.europeana.metis.sandbox.domain.RecordProcessEvent;
+import eu.europeana.metis.sandbox.service.record.RecordLogService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The type Harvest validation step.
+ */
+public class HarvestValidationStep implements ValidationStep {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private final RecordLogService recordLogService;
+    private ValidationStep nextValidationStep;
+
+    /**
+     * Instantiates a new Harvest validation step.
+     *
+     * @param recordLogService the record log service
+     */
+    public HarvestValidationStep(RecordLogService recordLogService) {
+        this.recordLogService = recordLogService;
+    }
+
+    @Override
+    public void setNextValidationStep(ValidationStep nextValidationStep) {
+        this.nextValidationStep = nextValidationStep;
+    }
+
+    @Override
+    public List<ValidationResult> validate(Record recordToValidate) {
+        List<ValidationResult> validationResults = new ArrayList<>();
+        try {
+            LOGGER.info("harvesting validation step virtual dataset {}", recordToValidate.getDatasetName());
+            validationResults.add(new ValidationResult(Step.HARVEST_FILE,
+                    new RecordValidationMessage(RecordValidationMessage.Type.INFO, "success"),
+                    ValidationResult.Status.PASSED));
+            validationResults.addAll(this.nextValidationStep.validate(recordToValidate));
```

Comment:
I see. Sort of recursion indeed, although it is handled well on the exceptions. It is unfolding all the way and then folding back to write the results in the list(s), where the performance concern is a legitimate one since there will be a lot of overhead on the stack(and that multiplied by concurrency).
I would say it it fine now, but we should leave a clearly visible todo message of this concern, so that if we in the future need to process more records we should then find a solution that mitigates those overheads.